rdt/transformers/base.py

```diff
@@ -1,28 +1,172 @@
+    in order to create a new transformer. The ``_fit`` method is optional,
     and ``fit_transform`` method is already implemented.
     """
 
-    def fit(self, data):
+    INPUT_TYPE = None
+    OUTPUT_TYPES = None
+    DETERMINISTIC_TRANSFORM = None
+    DETERMINISTIC_REVERSE = None
+    COMPOSITION_IS_IDENTITY = None
+    NEXT_TRANSFORMERS = None
+
+    @classmethod
+    def get_input_type(cls):
+        """Return the input type supported by the transformer.
+
+        Returns:
+            string:
+                Accepted input type of the transformer.
+        """
+        return cls.INPUT_TYPE
+
+    @staticmethod
+    def _add_prefix(dictionary, column_prefix):
+        # maybe add some validation of types here
+        if not dictionary:
+            return None
+
+        output = {}
+        for output_columns, output_type in dictionary.items():
+            output[f'{column_prefix}.{output_columns}'] = output_type
+
+        return output
+
+    def get_output_types(self):
+        """Return the output types supported by the transformer.
+
+        Returns:
+            dict:
+                Mapping from the transformed column names to supported data types.
+        """
+        return self._add_prefix(self._OUTPUT_TYPES, self._column_prefix)
+
+    def is_transform_deterministic(self):
+        """Return whether the transform is deterministic.
+
+        Returns:
+            bool:
+                Whether or not the transform is deterministic.
+        """
+        return self.DETERMINISTIC_TRANSFORM
+
+    def is_reverse_deterministic(self):
+        """Return whether the reverse transform is deterministic.
+
+        Returns:
+            bool:
+                Whether or not the reverse transform is deterministic.
+        """
+        return self.DETERMINISTIC_REVERSE
+
+    def is_composition_identity(self):
+        """Return whether composition of transform and reverse transform produces the input data.
+
+        Returns:
+            bool:
+                Whether or not transforming and then reverse transforming returns the input data.
+        """
+        return self.COMPOSITION_IS_IDENTITY
+
+    def get_next_transformers(self):
+        """Return the suggested next transformer to be used for each column.
+
+        Returns:
+            dict:
+                Mapping from transformed column names to the transformers to apply to each column.
+        """
+        return self._add_prefix(self._NEXT_TRANSFORMERS, self._column_prefix)
+
+    def fit(self, data, columns=None):
+        """Fit the transformer to the `columns` of the `data`.
+
+        Args:
+            data (pandas.DataFrame):
+                The entire table.
+            columns (list or str):
+                List or tuple of column names from the data to transform.
+                If only one column is provided, it can be passed as a string instead.
+        """
+        if columns is None:
```

Comment:
Thinking more about this, making `columns` optional does not seem necessary except for the backwards compatibility with the HyperTransformer, which is going to be changed immediately, so we can just make `columns` mandatory.